rioting-daisies/BangazonAPI
# Customer Resource Query String Parameter _include AND README.md

# Description

This pull request contains the implementation of the query string parameter `_include` which can accept two parameters. If the request is  `api/Customer?_include=products`, the request should return all customers with the list of products they are selling. If the request is `api/Customer/1?_include=products`, the response should be the specified customer (id: 1) along with all the products this customer is selling. If the request is `api/Customer?_include=payments`, the response should be all the customers along with all of their saved payment types. If the request is `api/Customer/1?_include=payments`, the response should be the specified customer (id: 1) and all of the payment types this customer has saved. 

Also, the query string parameter `q=[insert value to filter customers by]` can be used alongside the `_include` parameter.

There are also more tests to review that test these possible scenarios.

The README.md file is completed.

Fixes Issue # [(2)](https://github.com/rioting-daisies/BangazonAPI/issues/2)

## Type of change

- [x] This request requires a documentation update. 

# Testing Instructions

1. `git fetch --all`
2. `git checkout cbm-customerQueryString`
3. If you have already checked out this branch, `git pull origin cbm-customerQueryString` to update your local copy of the branch.
4. Review the code in the CustomerController.cs file within the GetAll and GetById
5. Review the tests within the TestCustomers.cs file for these new scenarios.
6. Review the code in Customer.cs to see that the Customer class now has a ListOfProducts property and a ListOfPaymentTypes property in order for this feature to work.
7. Review the readme.md file

# Checklist:
